hibernate-core/src/main/java/org/hibernate/query/programmatic/QuerySpecification.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.query.programmatic;
+
+import jakarta.persistence.criteria.Root;
+import org.hibernate.Incubating;
+import org.hibernate.query.restriction.Restriction;
+
+/**
+ * Commonality for all query specifications.
+ *
+ * @apiNote Query specifications only support a {@linkplain #getQueryRoot() single root}.
+ *
+ * @author Steve Ebersole
+ */
+@Incubating
+public interface QuerySpecification<T> {
+	/**
+	 * Get the root of the query.
+	 * E.g. given the HQL {@code "from Book"}, we have a single {@code Root<Book>}.
+	 */
+	Root<T> getQueryRoot();
+
```

Comment:
```suggestion

CommonQueryContract createQuery();

```

Reply:
Yeah, I had considered this.  I just did not see the purpose, kind of along the lines of our discussion from yesterday.  But its not hurting anything, and if it helps something... ok.

Reply:
It's not *super*-useful, but there's no cost in adding it and it could in principle be useful in some sorts of generic code.